tests/tracking/test_rest_tracking.py

```diff
@@ -544,6 +559,40 @@ def test_validate_path_is_safe_bad(path):
         validate_path_is_safe(path)
 
 
+@pytest.mark.skipif(not is_windows(), reason="This test only passes on Windows")
+@pytest.mark.parametrize(
+    "path",
+    [
+        "../path",
```

Comment:
some additional Windows-specific attack vectors:

```

    r'../etc/passwd',
    r'..\Windows\System32\',
    r'C:\Windows\System32\',
    r'/etc/passwd',
    r'::Windows\System32',
    r'..\..\..\..\Windows\System32\',
    r'../Windows/System32',
    r'.../...//',
    r'....\\',
    r'\\?\C:\Windows\System32\',
    r'\\.\C:\Windows\System32\',
    r'\\UNC\Server\Share\',
    r'\\Server\Share\folder\',
    r'\\127.0.0.1\c$\Windows\',
    r'\\localhost\c$\Windows\',
    r'\\smbserver\share\path\',
    r'..\\?\C:\Windows\System32\',
    r'file:///C:/Windows/System32/',
    r'C:/Windows/../Windows/System32/',
    r'C:\Windows\..\Windows\System32\',
    r'/Windows/no/such/path',
    r'../../../../../../../../../../../../Windows/System32',
    r'../../../../../../../../../../../../etc/passwd',
    r'../../../../../../../../../../../../var/www/html/index.html',
    r'../../../../../../../../../../../../usr/local/etc/openvpn/server.conf',
    r'../../../../../../../../../../../../Program Files (x86)',
    r'/../../../../../../../../../../../../Windows/System32',
    r'file:///etc/passwd',
    r'file:///d:/windows/repair/sam',
    r'file:///proc/version',
    r'file:///inetpub/wwwroot/global.asa',
    r'file://c:/boot.ini',
    r'/file://etc/passwd',
    r'/file:c:/passwd',
    r'/file://d:/windows/win.ini',
    r'/file://./windows/win.ini',
    r'/file://../windows/win.ini',
    r'/Windows\../etc/passwd',
    r'/Windows\..\Windows\System32\',
    r'/etc/passwd%00.jpg',
    r'/etc/passwd%00.html',
    r'/etc/passwd%00.txt',
    r'/etc/passwd%00.php',
    r'/etc/passwd%00.asp',
    r'/Windows\..\Windows\System32\cmd.exe',
    r'/Windows\..\Windows\System32\msconfig.exe',
    r'/Windows\..\Windows\System32\regedit.exe',
    r'/Windows\..\Windows\System32\taskmgr.exe',
    r'/Windows\..\Windows\System32\control.exe',
    r'/Windows\..\Windows\System32\services.msc',
    r'/Windows\..\Windows\System32\diskmgmt.msc',
    r'/Windows\..\Windows\System32\eventvwr.msc',
```

Reply:
<img width="738" alt="image" src="https://github.com/mlflow/mlflow/assets/82044803/619db23e-2bf1-420f-9402-02af20a7acd7">

I thought paths beginning with /xxx is always considered as a relative path on windows right? So it should be fine on windows? like "/path", it should be a good path.

Reply:
We can't believe what ChatGPT says.

Reply:
I'm just curious why /etc/passwd is not a good path in windows if it's a relative path?

Reply:
If it's a relative path, I think it's a good path.

Reply:
On windows, is it possible to join `/etc/passwd` and another path?

Reply:
I think so, path joining is regardless of relative or absolute path?